src/components/VioletEnroll/VioletEnrollModal.tsx

```diff
@@ -0,0 +1,79 @@
+import { useWeb3React } from '@web3-react/core'
+import useDebounce from 'hooks/useDebounce'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import { useIsUserRegisteredWithViolet } from 'state/registration/hooks'
+import { updateRegistrationState } from 'state/registration/reducer'
+import { useIsRegisteredWithViolet } from 'utils/temporary/useIsRegistered'
+
+import Modal from '../Modal'
+import VioletEnroll from './VioletEnroll'
+
+export default function VioletEnrollModal() {
```

Comment:
Between this and VioletEnroll component it got a bit confusing, maybe we should rename that one to VioletEnrollModalContent?